apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -1,502 +1,507 @@
+            }
+            if (_withMetaData.Contains(MetaDataType.UserName))
+            {
+                labels.Add(nameof(MetaDataType.UserName), loggingEvent.UserName ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.Domain))
+            {
+                labels.Add(nameof(MetaDataType.Domain), loggingEvent.Domain ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.LoggerName))
+            {
+                labels.Add(nameof(MetaDataType.LoggerName), loggingEvent.LoggerName ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.Level))
+            {
+                labels.Add(nameof(MetaDataType.Level), loggingEvent.Level?.Name ?? unknown);
+            }
+            TryAddGitRevisionId(labels);
+            foreach (var customLabel in _customLabels)
+            {
+                labels.Add(customLabel.Key, customLabel.Value);
+            }
+            var logEntry = new LogEntry
+            {
+                TextPayload = RenderLoggingEvent(loggingEvent),
+                Severity = s_levelMap[loggingEvent.Level],
+                Timestamp = loggingEvent.TimeStamp.ToTimestamp(),
+                LogName = _logName,
+                Resource = _resource,
+                Labels = { labels },
+            };
+            if (sourceLocation != null)
+            {
+                logEntry.SourceLocation = sourceLocation;
+            }
+            return logEntry;
+        }
+
+        private void TryAddGitRevisionId(Dictionary<string, string> labels)
+        {
+            try
+            {
+                var gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+                if (!String.IsNullOrWhiteSpace(gitId))
+                {
+                    labels.Add(SourceContext.GitRevisionIdLogLabel, gitId);
+                }
+            }
+            catch (Exception ex) when (
+                ex is SecurityException 
+                || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException 
+                || ex is UnauthorizedAccessException)
+            {
+                // This is best-effort only, exceptions from reading/parsing the source_context.json are ignored.
+            }
+        }
+
+        private LogEntrySourceLocation BuildLogEntryLocation(LoggingEvent loggingEvent)
+        {
+            string file = null;
+            long? line = null;
+            string function = null;
+            if (loggingEvent.LocationInformation?.FileName != null)
+            {
+                file = loggingEvent.LocationInformation?.FileName;
+            }
+            if (long.TryParse(loggingEvent.LocationInformation?.LineNumber, out long lineNumber))
+            {
+                line = lineNumber;
+            }
+            string function0 = null;
+            string fullTypeName = loggingEvent.LocationInformation?.ClassName;
+#if NET45

```

Comment:
Sorry, I wasn't clear.
Please put this `#if` back to where you had it before, within the `try` block; and use `#elif`, so the contents of the `try` block should look like this:
```
try
{
#if NET45
    return AppDomain.CurrentDomain.GetAssemblies()
        .SelectMany(a => a.GetTypes())
        .FirstOrDefault(t => t.FullName == fullTypeName);
#elif NETSTANDARD1_5
    // TODO: Support type lookup in netstandard
    return null;
#else
#error Unsupported platform.
#endif
}
catch
...
```
This puts the `#if` around the least amount of code.
And explicitly generates a compile-time error if we try change targeted platforms in any way, meaning we won't end up with any nasty surprises.
Thanks.